docs/proposals/job_controller.md

```diff
@@ -0,0 +1,192 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pods that require to run:
+* once on demand or at a specified point in time
```

Comment:
I'll be separating this into two objects, after talking to @derekwaynecarr yesterday, which was my original design some time ago, but foolishly I was convinced to merge that into one.